lib/eslint.js

```diff
@@ -620,9 +620,11 @@ module.exports = (function() {
      *      If this is not set, the filename will default to '<input>' in the rule context.
      * @param {boolean=} saveState Indicates if the state from the last run should be saved.
      *      Mostly useful for testing purposes.
+     * @param {boolean=} disableInlineConfig Disables inline comments' ability to change config once it is set.
+     *      Useful if you want to validate JS without comments overriding rules.
      * @returns {Object[]} The results as an array of messages or null if no messages.
      */
-    api.verify = function(textOrSourceCode, config, filename, saveState) {
+    api.verify = function(textOrSourceCode, config, filename, saveState, disableInlineConfig) {
```

Comment:
I think we can accept a `options` object here which contains all the current attributes plus the new attribute.
T make it backward compatible, we can check if the first argument is a `string` or `instance of SourceCode` then consider it a old style otherwise new style(which is a object).